heat/core/statistics.py

```diff
@@ -1100,6 +1132,231 @@ def mpi_argmin(a, b, _):
+    With this option, the result can broadcast correctly against the original array a.
+
+    Returns:
+    --------
+
+    percentile : process-local ht.tensor
+
+    """
+
+    def local_percentile(data, axis, indices):
+        """
+        Process-local percentile calculations
+
+        Input:
+        ------
+        data : ht.tensor
+        axis : int or TODO tuple of ints
+        indices : torch.tensor
+
+        Returns:
+        --------
+        percentile, torch.tensor, process-local
+        """
+
+        axis_slice = data.numdims * (slice(None, None, None),)
+        if indices.dtype == torch.long or indices.dtype == torch.int:
+            # interpolation 'lower', 'higher', or 'nearest'
+            axis_slice = axis_slice[:axis] + (indices.tolist(),) + axis_slice[axis + 1 :]
+            percentile = data[axis_slice]
+        else:
+            floor_indices = torch.floor(indices).type(torch.long)
+            axis_slice = axis_slice[:axis] + (floor_indices.tolist(),) + axis_slice[axis + 1 :]
+            lows = data._DNDarray__array[axis_slice]
+            ceil_indices = floor_indices + 1.0  # this one might spill over into next process: halo
+            axis_slice = axis_slice[:axis] + (ceil_indices.tolist(),) + axis_slice[axis + 1 :]
+            if ceil_indices.max().item() == data.lshape[axis]:
+                if axis == data.split:
+                    highs = data.array_with_halos[axis_slice]
+                else:
+                    # max percentile is 100.0
+                    ceil_indices[ceil_indices.argmax()] -= 1
+                    axis_slice = (
+                        axis_slice[:axis] + (ceil_indices.tolist(),) + axis_slice[axis + 1 :]
+                    )
+                    highs = data._DNDarray__array[axis_slice]
+            else:
+                highs = data._DNDarray__array[axis_slice]
+            # from here on, local (torch) operations
+            weights_shape = data.numdims * (1,)
+            weights_shape = weights_shape[:axis] + (indices.shape[0],) + weights_shape[axis + 1 :]
+            weights = torch.sub(
+                indices.reshape(weights_shape), torch.floor(indices).reshape(weights_shape)
+            )
+
+            percentile = lows + weights * (torch.sub(highs, lows))
+
+            if axis != 0:
+                # permute to have the number of percentiles at dimension 0
+                dims = tuple(range(percentile.ndim))
+                permute_dims = (axis,) + dims[:axis] + dims[axis + 1 :]
+                percentile = percentile.permute(permute_dims)
+
+        return percentile
+
+    # sanitize input
+    if not isinstance(x, dndarray.DNDarray):
+        raise TypeError("expected x to be a ht.DNDarray, but was {}".format(type(x)))
+    if isinstance(axis, list) or isinstance(axis, tuple):
+        raise NotImplementedError("ht.percentile(), tuple axis not implemented yet")
+
+    if isinstance(q, dndarray.DNDarray):
```

Comment:
It's an array_like of floats in numpy's implementation. I agree that it most cases it will be passed as a list or tuple of floats. However, one might have a `q = np.array(x,y,z)` that they pass to their numpy percentile calculations, and they'll just swap np with ht. I'd like the option to be available.

Reply:
I moved the check for tensor q to the last place in the if... elif... series, so in most cases it will just be skipped.

Reply:
I see :) Took me a while to understand the workflow